Wabbajack.Lib/MO2Installer.cs

```diff
@@ -178,27 +194,28 @@ private void AskToEndorse()
                 mods[b] = tmp;
             }
 
-            mods.PMap(Queue, mod =>
+            await mods.PMap(Queue, async mod =>
             {
-                var er = new NexusApiClient().EndorseMod(mod);
+                var client = await NexusApiClient.Get();
+                var er = await client.EndorseMod(mod);
                 Utils.Log($"Endorsed {mod.GameName} - {mod.ModID} - Result: {er.message}");
             });
             Info("Done! You may now exit the application!");
         }
 
-        private void BuildBSAs()
+        private async Task BuildBSAs()
         {
             var bsas = ModList.Directives.OfType<CreateBSA>().ToList();
             Info($"Building {bsas.Count} bsa files");
 
-            bsas.Do(bsa =>
```

Comment:
Could potentially be refactored to a Task.WhenAll if we want these to be done in parallel.  Not sure if that's desired or not, so I left it serial.